File: order-service/backend/src/main/java/be/kdg/sa/backend/domain/restaurant/Dish.java
package be.kdg.sa.backend.domain.restaurant;

import org.jmolecules.ddd.annotation.ValueObject;

import java.math.BigDecimal;

@ValueObject
public record Dish(
        DishId dishId,
        String name,
        String description,
        BigDecimal price,
        boolean vegetarian,
        boolean vegan,
        boolean glutenFree,
        String category,
        String status
) {
}

File: order-service/frontend/src/domain/dish.ts
export type DishCategory = "STARTER" | "MAIN" | "DESERT";
export type DishStatus = "AVAILABLE" | "SOLD_OUT"

export interface Dish {
    id: string;
    name: string;
    description?: string;
    price: number;
    vegetarian: boolean;
    vegan: boolean;
    glutenFree: boolean;
    category: DishCategory;
    status: DishStatus;
}